server/hwapi/endpoints/certification/logic.py

```diff
@@ -26,25 +26,27 @@
 
 
 def find_main_hardware_components(
-    db: Session, board_data: BoardValidator, bios_data: BiosValidator
-) -> tuple[models.Device, models.Bios]:
+    db: Session, board_data: BoardValidator, bios_data: BiosValidator | None
+) -> tuple[models.Device, models.Bios | None]:
     """
     A function to get "main hardware components" like board and bios. Can be extended
     in future
     """
     board = repository.get_board(
         db, board_data.manufacturer, board_data.product_name, board_data.version
     )
-    bios = repository.get_bios(
-        db, bios_data.vendor, bios_data.version, bios_data.firmware_revision
-    )
-    if not board or not bios:
+    if not board:
         raise ValueError("Hardware not certified")
-    return board, bios
+    if bios_data:
+        bios = repository.get_bios(
+            db, bios_data.vendor, bios_data.version, bios_data.firmware_revision
+        )
+        return board, bios
```

Comment:
if `bios_data` is not `None` shouldn't we still check that we found it in the repository and otherwise raise `ValueError("Hardware not certified")`

Reply:
you're right about that, I'll check it